case_studies/coadds/coadd_decoder.py

```diff
@@ -0,0 +1,299 @@
+    """Dataset of coadd galsim blends."""
+
+    def __init__(
+        self,
+        prior: CoaddUniformGalsimGalaxiesPrior,
+        decoder: FullCatalogDecoder,
+        background: ConstantBackground,
+        tile_slen: int,
+        max_sources_per_tile: int,
+        num_workers: int,
+        batch_size: int,
+        n_batches: int,
+        num_dithers: int,
+        fix_validation_set: bool = False,
+        valid_n_batches: Optional[int] = None,
+    ):
+        super().__init__(
+            prior,
+            decoder,
+            background,
+            tile_slen,
+            max_sources_per_tile,
+            num_workers,
+            batch_size,
+            n_batches,
+            fix_validation_set,
+            valid_n_batches,
+        )
+        self.num_dithers = num_dithers
+        self.slen = self.decoder.slen
+        self.pixel_scale = self.decoder.single_decoder.pixel_scale
+
+    def _sample_full_catalog(self):
+        params_dict = self.prior.sample(self.num_dithers)
+        dithers = params_dict["dithers"]
+        params_dict.pop("dithers")
+        params_dict["plocs"] = params_dict["locs"] * self.slen
+        params_dict.pop("locs")
+        params_dict = {k: v.unsqueeze(0) for k, v in params_dict.items()}
+        return FullCatalog(self.slen, self.slen, params_dict), dithers
+
+    def _get_images(self, full_cat, dithers):
+        size = self.slen + 2 * self.bp
+        noiseless, noiseless_centered, noiseless_uncentered, image0 = self.decoder.render_catalog(
+            full_cat=full_cat, psf=self.decoder.single_decoder.psf, dithers=dithers
+        )
+        #        image0 = rearrange(image0, "1 1 h w -> h w")
+
+        aligned_images = align_single_exposures(
+            image0.reshape(size, size), noiseless, size, dithers
+        )
+
+        background = self.background.sample(rearrange(aligned_images, "d h w -> d 1 h w").shape)
+
+        aligned_images = rearrange(aligned_images, "d h w -> d 1 h w")
+
+        weight = 1 / (aligned_images + background.clone().detach())
+
+        noisy_aligned_image = _add_noise_and_background(aligned_images, background)
+
+        coadded_image = _linear_coadd(noisy_aligned_image, weight)
+
+        return (
+            noiseless,
+            noiseless_centered,
+            noiseless_uncentered,
+            background,
+            coadded_image,
+        )
+
+    def _add_metrics(
+        self,
+        full_cat: FullCatalog,
+        noiseless: Tensor,
+        background: Tensor,
+        coadded_image: Tensor,
+    ):
+        return full_cat
+
+    def __getitem__(self, idx):
+        full_cat, dithers = self._sample_full_catalog()
+        (
+            noiseless,
+            noiseless_centered,
+            noiseless_uncentered,
+            background,
+            coadded_image,
+        ) = self._get_images(full_cat, dithers)
+        full_cat = self._add_metrics(full_cat, noiseless, background, coadded_image)
```

Comment:
You forgot the `tile_params` in the original dataset: 

```suggestion
        full_cat = self._add_metrics(full_cat, noiseless, background, coadded_image)
        tile_params = self._get_tile_params(full_cat)
```